content/token22-in-the-client.md

```diff
@@ -0,0 +1,470 @@
+---
+title: Supporting Token Extension Program from a Client
+objectives:
+- Learn how to effectively integrate multiple Solana token programs within client applications
+- Become proficient in utilizing the SPL TypeScript library for comprehensive token operations
+---
+
+# Summary
+
+- The `Token Extension Program` has all of the same functions as the `Token Program`, with added `extensions`
+- These two token programs: `Token Program` and `Token Extension Program` use seperate addresses and are not directly compatable
+- Supporting both requires simply specifying the correct program ID in client-side functions
+- By default, the SPL program library uses the original **`Token Program`** unless another is specified
+- The `Token Extension Program` may also be referred to as `Token22`
+
+# Overview
+
+The `Token Extensions Program` enhances the original `Token Program` by incorporating additional features known as extensions. These extensions are designed to address specific scenarios that previously necessitated developers to fork and alter the Solana Program Library, leading to split ecosystems and challenges in adoption. The introduction of the Token Extensions Program allows for these scenarios to be effectively handled.
+
+Since the `Token Program` and `Token Extensions Program` are different onchain programs, they are not interoperable. For example, a token minted with `Token Extensions Program` may not be transferred with the `Token Program`. As a result, we'll have to support both programs in our client-side applications. This means we'll want to explicitly handle mints from both the original Token Program (`TOKEN_PROGRAM_ID`) and the Extension Program (`TOKEN_2022_PROGRAM_ID`).
+
+Fortunately, the interfaces for the two programs remain consistent, allowing the use of `spl-token` helper functions in either program by simply swapping the program ID (the function uses the original Token Program by default if no program ID is provided). Most of the time, end users are not concerned with the specific token program being used. As such, implementing additional logic to track, assemble, and merge details from both token varieties is essential to guarantee a smooth user experience.
+
+Lastly, the Token Extension Program is internally named "Token 22" as in `TOKEN_2022_PROGRAM_ID`. They are the same thing.
```

Comment:
Maybe call out the spec name thing here. "Token 22 is often used as the technical name. If you see someone refer to the Token 22 Program, they are referring to the Token Extensions Program."